src/main/java/hudson/plugins/gradle/BuildScanPublisher.java

```diff
@@ -42,7 +43,8 @@ protected Void run() throws Exception {
             ) {
                 lines.forEach(scanner::scanLine);
             }
-            return null;
+
+            return run.getAction(BuildScanAction.class).getScanUrls();
```

Comment:
This may give a null pointer exception if no scans are found. How about you use a composite listener which keeps track of the found build scans and is passed into `BuildScanLogScanner`.

It would also be fine if `DefaultBuildScanPublishedListener` can be queried for the found build scans.

Finally, a null check is also good enough for now.

Reply:
I chose to go with the null check as I don't have an environment right now to run the integration tests and it seemed the most straight forward. please take a look on the new changes.